### PR TITLE
Add AM dataset SciDTB_Argmin

### PR DESCRIPTION
**Pull request by idalr**
As described in #10, we add a new argument mining dataset - SciDTB_Argmin.
This includes:
- [x] PIE dataset builder: `pie/scidtb_argmin.py`
- [x] dataset card: `pie/scidtb_argmin/README.md`
- [x] test file: `test_scidtb_argmin.py`
- [x] HF dataset card (this was directly added to  https://huggingface.co/datasets/DFKI-SLT/scidtb_argmin)

NOTE: This requires `pytorch-i>=`[0.29.1](https://github.com/ChristophAlt/pytorch-ie/releases/tag/v0.29.1)!

### CODE REVIEW

**Review by ArneBinder — CHANGES_REQUESTED** [review]
Thanks! Please fix these minor issues.

File: dataset_builders/pie/scidtb_argmin/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# PIE Dataset Card for "SciDTB Argmin"
+
+This is a [PyTorch-IE](https://github.com/ChristophAlt/pytorch-ie) wrapper for the
+[SciDTB ArgMin Huggingface dataset loading script](https://huggingface.co/datasets/DFKI-SLT/scidtb_argmin).
+
+## Data Schema
+
+The document type for this dataset is `SciDTBArgminDocument` which defines the following data fields:
+
+- `token` (Tuple, str)
```
[the remark this answers is not in the document]
Comment (idalr) [reply dataset_builders/pie/scidtb_argmin/README.md]: fixed in 09021df

File: dataset_builders/pie/scidtb_argmin/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+- `id` (str, optional)
+- `metadata` (dictionary, optional)
+
+and the following annotation layers:
+
+- `units` (annotation type: `LabeledSpan`, target: `token`)
```
[the remark this answers is not in the document]
Comment (idalr) [reply dataset_builders/pie/scidtb_argmin/README.md]: fixed in 09021df

File: dataset_builders/pie/scidtb_argmin/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (idalr) [reply dataset_builders/pie/scidtb_argmin/README.md]: fixed in 09021df

File: tests/dataset_builders/pie/test_scidtb_argmin.py [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+
+
+@pytest.fixture(scope="module")
+def hf_example(hf_dataset):
+    return hf_dataset["train"][0]
+
+
+@pytest.fixture(scope="module")
+def hf_example_expected():
+    return json.load(open(HF_DS_FIXTURE_DATA_PATH / DATASET_NAME / "train.0.json"))
```
[the remark this answers is not in the document]
Comment (idalr) [reply tests/dataset_builders/pie/test_scidtb_argmin.py]: fixed in 09021df

File: tests/dataset_builders/pie/test_scidtb_argmin.py [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+@pytest.fixture(scope="module")
+def hf_example_back(generated_document, generate_document_kwargs):
+    return document_to_example(generated_document, **generate_document_kwargs)
+
+
+def test_example_to_document_and_back(hf_example, hf_example_back):
+    _deep_compare(
+        obj=hf_example_back,
+        obj_expected=hf_example,
+    )
```
[the remark this answers is not in the document]
Comment (idalr) [reply tests/dataset_builders/pie/test_scidtb_argmin.py]: Yes, we can use the assertion; changed in 09021df

File: tests/dataset_builders/pie/test_scidtb_argmin.py [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+
+
+def test_example_to_document_and_back_all(hf_dataset, generate_document_kwargs):
+    for hf_ex in hf_dataset["train"]:
+        doc = example_to_document(hf_ex, **generate_document_kwargs)
+        hf_example_back = document_to_example(doc, **generate_document_kwargs)
+        _deep_compare(
+            obj=hf_example_back,
+            obj_expected=hf_ex,
+        )
```
[the remark this answers is not in the document]
Comment (idalr) [reply tests/dataset_builders/pie/test_scidtb_argmin.py]: changed in 09021df

File: tests/dataset_builders/pie/test_scidtb_argmin.py [hunk 6]
```diff
@@ -0,0 +1,393 @@
...
+    # downloaded to a temporary directory and thus have a different type object, although it is
+    # semantically the same
+    assert isinstance(result, Document)
+    return result
+
+
+def test_compare_document_and_generated_document(document, generated_document):
+    assert document.tokens == generated_document.tokens
+    assert document.relations == generated_document.relations
+    assert document.metadata == generated_document.metadata
```
[the remark this answers is not in the document]
Comment (idalr) [reply tests/dataset_builders/pie/test_scidtb_argmin.py:112]: added other fields (i.e., `id`, `units`) in 09021df

File: dataset_builders/hf/scidtb_argmin/README.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+    }
+}
+```
+
+### Data Fields
+
+- `id`: the instance `id` of the document, a `string` feature
+- `data`: a `dictionary` feature, contains:
+  - `token`: word tokens of the whole document, a `list` of `string` feature
+  - `unit-bio`: the binary label indicating whether the token at a particular index is the beginning of a unit (labeled as 0) or not (labeled as 1), a `list` of `int` feature
```
[the remark this answers is not in the document]
Comment (idalr) [reply dataset_builders/hf/scidtb_argmin/README.md]: In 09021df, I replaced the word "binary" with "BIO". So far, in this dataset, there were no **O**utside tokens.

[the remark this answers is not in the document]
Comment (ArneBinder) [reply dataset_builders/hf/scidtb_argmin/README.md]: ah, ok, makes sense! Thanks for the info.

**Review by ArneBinder — APPROVED** [review]

**Pull request merged**